### PR TITLE
goldendict-ng: initial upload

### PR DESCRIPTION
**Pull request by shenlebantongying**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

### CODE REVIEW

**Comment by classabbyamp** [comment]
commit message should be `New package: goldendict-ng-23.06.01`

File: srcpkgs/goldendict-ng/template [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+version=23.06.01
+revision=0
+build_style=cmake
+configure_args="-DUSE_SYSTEM_FMT=ON"
+hostmakedepends="pkg-config qt6-base"
+makedepends="qt6-base-devel qt6-tools-devel qt6-declarative-devel
+	qt6-webengine qt6-svg-devel qt6-qt5compat
+	libXtst-devel fmt hunspell-devel libeb-devel
+	libzim-devel libvorbis-devel ffmpeg-devel
+	lzo-devel zlib-devel bzip2-devel liblzma-devel libzstd-devel"
```
Comment (classabbyamp) [inline srcpkgs/goldendict-ng/template:12]: indent this with a single space
```suggestion
 qt6-webengine qt6-svg-devel qt6-qt5compat
 libXtst-devel fmt hunspell-devel libeb-devel
 libzim-devel libvorbis-devel ffmpeg-devel
 lzo-devel zlib-devel bzip2-devel liblzma-devel libzstd-devel"
```

File: srcpkgs/goldendict-ng/template [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+short_desc="A feature-rich dictionary lookup program"
+maintainer="shenlebantongying <shenlebantongying@gmail.com>"
+license="GPL-3.0-or-later"
+homepage="https://xiaoyifang.github.io/goldendict-ng/"
+distfiles="https://github.com/xiaoyifang/goldendict-ng/archive/refs/tags/v23.06.01-ChildrenDay.230601.6d3d4053.tar.gz"
```
Comment (classabbyamp) [inline srcpkgs/goldendict-ng/template:17]: use `$version` in the distfiles url

**Comment by shenlebantongying** [comment]
All fixed

**Comment by shenlebantongying** [comment]
Sadly, void linux doesn't have `qt6-speech` at the moment, but do have `qt5-speech`

**Comment by shenlebantongying** [comment]
The `libzim` is too old (from ~3 years ago.)

**Comment by shenlebantongying** [comment]
@classabbyamp The package is ready. I tested it locally, and it works fine.

Zim support is disabled. I will update `libzim` later.

image

**Comment by classabbyamp** [comment]
@shenlebantongying please rebase on master to let CI run again

File: srcpkgs/goldendict-ng/template [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+revision=1
+build_style=qmake
+configure_args="CONFIG+=chinese_conversion_support CONFIG+=use_iconv" 
+hostmakedepends="pkg-config qt5-qmake qt5-host-tools"
+makedepends="qt5-devel qt5-tools-devel qt5-declarative-devel qt5-webchannel-devel qt5-location-devel
+ qt5-webengine-devel qt5-svg-devel qt5-multimedia-devel qt5-speech-devel qt5-x11extras-devel
+ libXtst-devel hunspell-devel libeb-devel xapian-core-devel
+ libvorbis-devel ffmpeg-devel opencc-devel
+ lzo-devel zlib-devel bzip2-devel liblzma-devel libzstd-devel"
+short_desc="A feature-rich dictionary lookup program"
```
Comment (classabbyamp) [inline srcpkgs/goldendict-ng/template:13]:
```suggestion
short_desc="Feature-rich dictionary lookup program"
```

**Comment by shenlebantongying** [comment]
Thanks for review again, I will do it within this week.

**Comment by shenlebantongying** [comment]
Whoops, just messed up git :)

Qt5

```bash
# Template file for 'goldendict-ng'
pkgname=goldendict-ng
version=23.06.01
revision=1
build_style=qmake
configure_args="CONFIG+=chinese_conversion_support CONFIG+=use_iconv" 
hostmakedepends="pkg-config qt5-qmake qt5-host-tools"
makedepends="qt5-devel qt5-tools-devel qt5-declarative-devel qt5-webchannel-devel qt5-location-devel
 qt5-webengine-devel qt5-svg-devel qt5-multimedia-devel qt5-speech-devel qt5-x11extras-devel
 libXtst-devel hunspell-devel libeb-devel xapian-core-devel
 libvorbis-devel ffmpeg-devel opencc-devel
 lzo-devel zlib-devel bzip2-devel liblzma-devel libzstd-devel"
short_desc="A feature-rich dictionary lookup program"
maintainer="shenlebantongying <shenlebantongying@gmail.com>"
license="GPL-3.0-or-later"
homepage="https://xiaoyifang.github.io/goldendict-ng/"
distfiles="https://github.com/xiaoyifang/goldendict-ng/archive/refs/tags/v$version.tar.gz"
checksum=0aabbe6acb4f1f9f9b95d77066b9823a86647a7c0c58a85c04b86a246fc9429b
```

Qt6

```bash
# Template file for 'goldendict-mg'
pkgname=goldendict-ng
version=23.06.01
revision=0
build_style=cmake
configure_args="-DUSE_SYSTEM_FMT=ON"
hostmakedepends="pkg-config qt6-base"
makedepends="qt6-base-devel qt6-tools-devel qt6-declarative-devel
 qt6-webengine qt6-svg-devel qt6-qt5compat
 libXtst-devel fmt hunspell-devel libeb-devel
 libzim-devel libvorbis-devel ffmpeg-devel
 lzo-devel zlib-devel bzip2-devel liblzma-devel libzstd-devel"
short_desc="A feature-rich dictionary lookup program"
maintainer="shenlebantongying <shenlebantongying@gmail.com>"
license="GPL-3.0-or-later"
homepage="https://xiaoyifang.github.io/goldendict-ng/"
distfiles="https://github.com/xiaoyifang/goldendict-ng/archive/refs/tags/v$version.tar.gz"
checksum=0aabbe6acb4f1f9f9b95d77066b9823a86647a7c0c58a85c04b86a246fc9429b
```

**Comment by shenlebantongying** [comment]
Sadly, for some reason

**Comment by shenlebantongying** [comment]
@classabbyamp all things fixed :)

The CI still needs approval.

**Comment by classabbyamp** [comment]
should add `conflicts="goldendict>=0"`

**Comment by shenlebantongying** [comment]
The CI should pass now. 

`conflict` added

update check added

(The update check is a bit abnormal because 1. the software author has this obsession of adding weird names after version like "mooncake" 2. releasing test builds too often and cannot find the stable one from https://github.com/xiaoyifang/goldendict-ng/releases )

**Comment by shenlebantongying** [comment]
trailing whitespace fixed.

**Comment by shenlebantongying** [comment]
Can this be merged now? I don't see further problems.

Thanks for reviewing.

**Comment by shenlebantongying** [comment]
Can this be merged now? I don't see further problems.

Thanks for reviewing.

**Comment by shenlebantongying** [comment]
If someone else is interested, feel free to take this over.

**Comment by darkralts** [comment]
@shenlebantongying hey how do i compile this?

**Comment by darkralts** [comment]
i got it working really easy,, why was it never accepted? 
image


is it easy to update the package? change the version and the dist files in the template?

**Pull request closed without merging**